internal/global/trace.go

```diff
@@ -143,5 +144,44 @@ func (t *tracer) Start(ctx context.Context, name string, opts ...trace.SpanOptio
 	if delegate != nil {
 		return delegate.(trace.Tracer).Start(ctx, name, opts...)
 	}
-	return noop.Tracer.Start(ctx, name, opts...)
+
+	s := nonRecordingSpan{sc: trace.SpanContextFromContext(ctx)}
+	ctx = trace.ContextWithSpan(ctx, s)
+	return ctx, s
+}
+
+// nonRecordingSpan is a minimal implementation of a Span that wraps a
+// SpanContext. It performs no operations other than to return the wrapped
+// SpanContext.
+type nonRecordingSpan struct {
+	sc trace.SpanContext
 }
+
+var _ trace.Span = nonRecordingSpan{}
```

Comment:
```suggestion
var _ trace.Span = (*nonRecordingSpan)(nil)
```

Small nit, but this should save the allocation of a `nonRecordingSpan` that'll never be used.  Not sure if the compiler can optimize that away, but this is the form used in much of the rest of this repo for these interface satisfaction assertions.

Reply:
The method receiver for this type is a value not a pointer meaning this needs to evaluate the value not a pointer.